core/src/identity.rs

```diff
@@ -174,42 +173,37 @@ impl Keypair {
             }
         };
 
-        Ok(pk.encode_to_vec())
+        Ok(quick_protobuf::serialize_into_vec(&pk).expect("Encoding to succeed."))
     }
 
     /// Decode a private key from a protobuf structure and parse it as a [`Keypair`].
     pub fn from_protobuf_encoding(bytes: &[u8]) -> Result<Keypair, DecodingError> {
-        use prost::Message;
-
-        let mut private_key = keys_proto::PrivateKey::decode(bytes)
-            .map_err(|e| DecodingError::new("Protobuf").source(e))
-            .map(zeroize::Zeroizing::new)?;
+        let mut private_key: zeroize::Zeroizing<keys_proto::PrivateKey> =
+            quick_protobuf::deserialize_from_slice(bytes)
```

Comment:
Nit: I'd prefer using turbo-fish here.

```suggestion
        let mut private_key =
            quick_protobuf::deserialize_from_slice::<keys_proto::PrivateKey>(bytes)
```